Poker.BE/Poker.BE.Domain/Security/UserManager.cs

```diff
@@ -44,22 +50,28 @@ public class UserManager
             return false;
         }
 
-        public User LogIn (string userName, string password){
-            if (CheckExistingUser(userName)){ // We check that the user is existing in our DB
-               User UserToCheck;
-               if (UsersDictionary.TryGetValue(userName, out UserToCheck) ){ // We take the User Object from our DB
-                   string GoodPassword = UserToCheck.Password;
-                   bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
-                   if (arePasswordMatching) {
-                       UserToCheck.Connect();
-                       return UserToCheck;
-                    }
-               }
-            }
-            return null;
-        }
+		public User LogIn(string userName, string password)
+		{
+			if (!CheckExistingUser(userName)) // We check that the user is existing in our DB
+				throw new UserNotFoundException();
+			User UserToCheck;
+			if (UsersDictionary.TryGetValue(userName, out UserToCheck))
+			{ // We take the User Object from our DB
+				string GoodPassword = UserToCheck.Password;
+				bool arePasswordMatching = GoodPassword.Equals(password, StringComparison.Ordinal); // We check that the password is correct
```

Comment:
👎 add `// Note: ...` if you want to describe your code inside the code flow. and please try not to make too long source code lines.